src/vs/workbench/api/browser/mainThreadOutputService.ts

```diff
@@ -81,6 +81,14 @@ export class MainThreadOutputService extends Disposable implements MainThreadOut
 		return undefined;
 	}
 
+	public $replaceAll(channelId: string, till: number, value: string): Promise<void> | undefined {
+		const channel = this._getChannel(channelId);
```

Comment:
Needs to be guarded with the [`checkProposedApiEnabled `](https://github.com/microsoft/vscode/blob/27753407604003e199b270bb0ba504eef8c4beff/src/vs/workbench/services/extensions/common/extensions.ts#L322)-check

Reply:
The best place for this was actually here:

https://github.com/microsoft/vscode/blob/8e2f1765b604e601705be0a559acdda319fbf830/src/vs/workbench/api/common/extHostOutput.ts#L66

I had to refactor some host output classes as extension was not exposed to be able to call `checkProposedApiEnabled`.